File: component-factory/base/src/main/java/com/lazylite/mod/widget/textview/IconView.java
package com.lazylite.mod.widget.textview;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatTextView;

import com.lazylite.mod.utils.FontUtils;


public class IconView extends AppCompatTextView {

    public IconView(Context context) {
        this(context, null);
    }

    public IconView(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public IconView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        setTypeface(FontUtils.getInstance().getIconFontType(context));
    }
}

File: component-factory/UICompose/src/main/java/com/godq/compose/iconview/IconView.kt
package com.godq.compose.iconview

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView

class IconView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : AppCompatTextView(
    context, attrs, defStyleAttr
) {
    init {
        getIconFontType()?.apply {
            typeface = this
        }
    }
}